testing/tests/DevExpress.ui.widgets.form/form.validationRules.tests.js

```diff
@@ -0,0 +1,418 @@
+                },
+                validationResult: options.validationResult
+            });
+        });
+    };
+
+    runChangeRuleRageMaxTest({ fieldValue: 10, initialMax: 11, targetMax: 1, validationResult: false });
+    runChangeRuleRageMaxTest({ fieldValue: 10, initialMax: 1, targetMax: 11, validationResult: true });
+});
+
+QUnit.testInActiveWindow("Add RangeRule to item.validationRules", function(assert) {
+    const runSetRangeRuleTest = (options) => {
+        runChangeValidationRuleTest(extend(
+            {
+                assert,
+                fieldValue: 10,
+                newValidationRules: [{ type: "range", max: 1 }],
+                validationResult: false,
+            },
+            options
+        ));
+    };
+
+    [undefined, null, []].forEach(validationRules => {
+        [true, false].forEach(useItemOption => {
+            runSetRangeRuleTest({ validationRules, useItemOption, isKeepFocusSupported: false });
+        });
+    });
+});
+
+QUnit.testInActiveWindow("Remove RangeRule from item.validationRules", function(assert) {
+    const runRemoveRangedRuleTest = (options) => {
+        runChangeValidationRuleTest(extend(
+            {
+                assert,
+                fieldValue: 10,
+                validationRules: [{ type: "range", max: 1 }],
+                validationResult: true,
+            },
+            options
+        ));
+    };
+
+    [undefined, null, []].forEach(newValidationRules => {
+        runRemoveRangedRuleTest({ newValidationRules, useItemOption: false, isKeepFocusSupported: true });
+    });
+    [undefined, null, []].forEach(newValidationRules => {
+        runRemoveRangedRuleTest({ newValidationRules, useItemOption: true, isKeepFocusSupported: false });
+    });
+});
+
+QUnit.testInActiveWindow("Add RequiredRule to item.validationRules", function(assert) {
+    [undefined, null, []].forEach(validationRules => {
+        [true, false].forEach(useItemOption => {
+            runChangeValidationRuleTest({
+                assert,
+                fieldValue: null,
+                validationRules,
+                newValidationRules: [{ type: "required" }],
+                validationResult: false,
+                isKeepFocusSupported: false,
+                useItemOption
+            });
+        });
+    });
+});
+
+QUnit.testInActiveWindow("Remove RequiredRule from item.validationRules", function(assert) {
+    [undefined, null, []].forEach(newValidationRules => {
+        [true, false].forEach(useItemOption => {
+            runChangeValidationRuleTest({
+                assert,
+                fieldValue: null,
+                validationRules: [{ type: "required" }],
+                newValidationRules,
+                validationResult: true,
+                isKeepFocusSupported: false,
+                useItemOption
+            });
+        });
+    });
+});
+
+QUnit.testInActiveWindow("Change item.isRequired", function(assert) {
```

Comment:
You can write like this
`QUnit.testInActiveWindow("Change item.isRequired", assert => {`

Reply:
I will double check this in my new tests